lib/td/client/api.rb

```diff
@@ -121,7 +121,12 @@ def self.normalized_msgpack(record, out = nil)
         record[k] = v.to_s
       end
     }
-    record.to_msgpack(out)
+    if out
+      out << record.to_msgpack
+      out
+    else
+      record.to_msgpack
+    end
```

Comment:
Feeding output buffer as an argument value for `#to_msgpack` is very old API for msgpack-ruby (for v0.4.x) and it's obsolete now (v0.5.x or later), for both of CRuby/JRuby.
That argument should be an instance of Packer.